jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -535,10 +536,10 @@ def _initialize_model_component(self, model_comp, comp_label, poly_order=None):
             init_x = masked_spectrum.spectral_axis
             init_y = masked_spectrum.flux
 
-        # equivs for spectral density and flux<>flux/pix2. revisit
-        # when generalizing plugin UC equivs.
+        # equivs for spectral density and flux<>flux/pix2
         equivs = _eqv_flux_to_sb_pixel() + u.spectral_density(init_x)
-        init_y = init_y.to(self._units['y'], equivs)
+
+        init_y = flux_conversion_general([init_y.value], init_y.unit, self._units['y'], equivs)
```

Comment:
Leaving another note here to retest model fitting in Specviz2d following merging this PR (#3253). Bug was preexisting unit conversion plugin but still worth retesting.